cloudinit/sources/DataSourceAzure.py

```diff
@@ -559,9 +561,19 @@ def _get_data(self):
                 logfunc=LOG.debug, msg='Crawl of metadata service',
                 func=self.crawl_metadata
             )
-        except sources.InvalidMetaDataException as e:
-            LOG.warning('Could not crawl Azure metadata: %s', e)
+        except Exception as e:
+            report_diagnostic_event(
+                'Could not crawl Azure metadata: %s' % e,
+                logger_func=LOG.error)
+            try:
+                self._report_failure(
+                    description=DEFAULT_REPORT_FAILURE_USER_VISIBLE_MESSAGE)
+            except Exception as inner_e:
```

Comment:
`self._report_failure` catches (and does not re-raise) every `Exception` that it could raise; I don't think this handler can ever be hit.

Reply:
Removed.